client/lib/application/src/folded-scope-list.jsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import _ from 'lodash';
+import Icon from 'react-fa';
+import 'common/asset/less/application/scope-list.less';
+
+class OwnerScopeList extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            allResources: props.allResources,
+            foldings:{},
+            term: '',
+            filtered: props.scopes,
+            selected: props.selected || []
+        };
+
+      }
+
+      componentWillMount(){
+        
+        this.state.selected.map(selected => {
+          if (!this.state.foldings || !this.state.foldings[selected.resource_type_id]) this.toggleFolding(selected.resource_type_id);
+        });
+      }
+
+    componentWillReceiveProps(nextProps) {
+        if (!this.state.term.length) {
+            this.setState({
+                filtered: nextProps.scopes,
+                selected: nextProps.selected,
+                allResources: nextProps.allResources
+            });
+        }
+    }
+
+    toggleSelection(scope) {
+        let scopeToAdd = {
+                scope_id: scope.id,
+                resource_type_id: scope.resource_type_id
+            },
+            idx = _.findLastIndex(this.state.selected, scopeToAdd);
+
+        if (idx < 0) {
+            this.state.selected.push(scopeToAdd);
```

Comment:
I think the state object shouldn't be mutated directly.

Reply:
i've get rid of this code by extending my scope-list superclass with my class. So technically this problem still persists but i don't multiply this bug. And resolving of this bug is out of scope of current ticket